server/graphql/v1/mutations/collectives.js

```diff
@@ -710,3 +711,222 @@ export async function unarchiveCollective(_, args, req) {
 
   return collective.update({ isActive: true, deactivatedAt: null });
 }
+
+export async function deleteCollective(_, args, req) {
+  if (!req.remoteUser) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in to delete a collective',
+    });
+  }
+
+  const collective = await models.Collective.findByPk(args.id);
+  if (!collective) {
+    throw new errors.NotFound({
+      message: `Collective with id ${args.id} not found`,
+    });
+  }
+
+  if (!req.remoteUser.isAdmin(collective.id)) {
+    throw new errors.Unauthorized({
+      message: 'You need to be logged in as an Admin.',
+    });
+  }
+
+  const transactionCount = await models.Transaction.count({
+    where: {
+      [Op.or]: [{ CollectiveId: collective.id }, { FromCollectiveId: collective.id }],
+    },
+  });
+  const orderCount = await models.Order.count({
+    where: {
+      [Op.or]: [{ CollectiveId: collective.id }, { FromCollectiveId: collective.id }],
+    },
+  });
+
+  if (transactionCount > 0 || orderCount > 0) {
+    throw new Error('Can not delete collective with existing orders.');
+  }
+
+  const expenseCount = await models.Expense.count({
+    where: { CollectiveId: collective.id, status: 'PAID' },
+  });
+
+  if (expenseCount > 0) {
+    throw new Error('Can not delete collective with paid expenses.');
+  }
+
+  return models.Member.findAll({
+    where: { CollectiveId: collective.id },
+  })
+    .then(members => {
+      return map(
+        members,
+        member => {
+          return member.destroy();
+        },
+        { concurrency: 3 },
+      );
+    })
+    .then(() => debugDelete('deleteCollectiveMembers'))
+    .then(async () => {
+      const expenses = await models.Expense.findAll({
```

Comment:
Doing `models.Expenses.destroy({ where: { CollectiveId: collective.id } })` would have been much more efficient and concise.